data/publisher/src/main/java/tech/pegasys/teku/data/publisher/MetricsDataFactory.java

```diff
@@ -54,6 +55,152 @@ public MetricsDataFactory(MetricsSystem metricsSystem) {
     return metricList;
   }
 
+  private BaseMetricData extractSystemData(
+      final Map<String, Object> values, final TimeProvider timeProvider) {
+
+    final Long cpuNodeSystemSecondsTotal;
+    final Long cpuNodeUserSecondsTotal;
+
+    final Long memoryNodeBytesTotal;
+    final Long memoryNodeBytesFree;
+    final Long memoryNodeBytesCached;
+    final Long memoryNodeBytesBuffers;
+
+    final Long diskNodeBytesTotal;
+    final Long diskNodeBytesFree;
+    final Long diskNodeIoSeconds;
+    final Long diskNodeReadsTotal;
+    final Long diskNodeWritesTotal;
+
+    final Long networkNodeBytesTotalReceive;
+    final Long networkNodeBytesTotalTransmit;
+    final Long miscNodeBootTsSeconds;
+
+    if (values.containsKey("cpu_seconds_total")) {
+      cpuNodeSystemSecondsTotal =
+          ((Double) ((Observation) values.get("cpu_seconds_total")).getValue()).longValue();
+    } else {
+      cpuNodeSystemSecondsTotal = null;
+    }
```

Comment:
we can probably encapsulate this logic in a function to reduce the rewriting... might need more than 1 version, but basically then the variables are all set from the same code and cutting down a lot on the duplication... then can just do
```
cpuNodeSystemSecondsTotal = getValue("cpu_seconds_total");
```

also the casting seems fairly confused, we're casting a long value to a double, then assigning it into a long?